cmd/publish.go

```diff
@@ -89,6 +89,7 @@ var publishCmd = &cobra.Command{
 			}).Info("Publishing service")
 			go publisher.Publish(ip, iface, service, shutdownChannel, waitGroup)
 		}
+		go publisher.IfaceCheck(ip, iface)
```

Comment:
I'm new to GOLANG, so I'm not sure if it's a good idea.
Do you think that changing [1] to something similar to what we have in [2] , could be useful?

I mean something like that:

for {
		select {
		case <-shutdown:
			'what we have do incase of shutdown'
		default:
                      //call to any periodic function ( IfaceCheck, and we can add other functions in future)
                     IfaceCheck()
                     time.sleep(X)
}


[1] https://github.com/openshift/mdns-publisher/blob/cb9c32caaa13e3ee63a2f19a706b05257212b175/pkg/publisher/publisher.go#L35
[2] https://github.com/openshift/baremetal-runtimecfg/blob/master/pkg/monitor/monitor.go#L49

Reply:
Hmm, I wasn't familiar with that pattern in Go, but I think we can use it to eliminate the pid stuff in the check function. I'm not sure we want to get rid of the goroutine, but we can use another channel to do the communication instead of signaling. I'll include that change in the next version of this patch.